consensus/spos/worker.go

```diff
@@ -23,23 +23,24 @@ import (
 
 // Worker defines the data needed by spos to communicate between nodes which are in the validators group
 type Worker struct {
-	consensusService   ConsensusService
-	blockChain         data.ChainHandler
-	blockProcessor     process.BlockProcessor
-	bootstrapper       process.Bootstrapper
-	broadcastMessenger consensus.BroadcastMessenger
-	consensusState     *ConsensusState
-	forkDetector       process.ForkDetector
-	keyGenerator       crypto.KeyGenerator
-	marshalizer        marshal.Marshalizer
-	hasher             hashing.Hasher
-	rounder            consensus.Rounder
-	shardCoordinator   sharding.Coordinator
-	singleSigner       crypto.SingleSigner
-	syncTimer          ntp.SyncTimer
-	headerSigVerifier  RandSeedVerifier
-	appStatusHandler   core.AppStatusHandler
-	chainID            []byte
+	consensusService        ConsensusService
+	blockChain              data.ChainHandler
+	blockProcessor          process.BlockProcessor
+	bootstrapper            process.Bootstrapper
+	broadcastMessenger      consensus.BroadcastMessenger
+	consensusState          *ConsensusState
+	forkDetector            process.ForkDetector
+	keyGenerator            crypto.KeyGenerator
+	marshalizer             marshal.Marshalizer
+	hasher                  hashing.Hasher
+	rounder                 consensus.Rounder
+	shardCoordinator        sharding.Coordinator
+	singleSigner            crypto.SingleSigner
+	syncTimer               ntp.SyncTimer
+	headerSigVerifier       RandSeedVerifier
+	headerIntegrityVerifier HeaderIntegrityVerifier
+	appStatusHandler        core.AppStatusHandler
+	chainID                 []byte
```

Comment:
no more need for chain ID in consensus. it can be held by the header integrity validity component.

Reply:
It is used to check a consensus message which is different to a header